src/main/java/thePackmaster/patches/PacksRunHistoryPatch.java

```diff
@@ -0,0 +1,87 @@
+package thePackmaster.patches;
+
+import basemod.ReflectionHacks;
+import com.badlogic.gdx.graphics.g2d.SpriteBatch;
+import com.evacipated.cardcrawl.modthespire.lib.*;
+import com.evacipated.cardcrawl.modthespire.patcher.PatchingException;
+import com.megacrit.cardcrawl.core.CardCrawlGame;
+import com.megacrit.cardcrawl.dungeons.AbstractDungeon;
+import com.megacrit.cardcrawl.metrics.Metrics;
+import com.megacrit.cardcrawl.monsters.MonsterGroup;
+import com.megacrit.cardcrawl.screens.runHistory.RunHistoryScreen;
+import com.megacrit.cardcrawl.screens.stats.RunData;
+import javassist.*;
+import thePackmaster.SpireAnniversary5Mod;
+import thePackmaster.ThePackmaster;
+import thePackmaster.packs.AbstractCardPack;
+
+import java.lang.reflect.Field;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class PacksRunHistoryPatch {
+    private static final String[] TEXT = CardCrawlGame.languagePack.getUIString(SpireAnniversary5Mod.makeID("RunHistory")).TEXT;
+
+    @SpirePatch(clz = CardCrawlGame.class, method = SpirePatch.CONSTRUCTOR)
+    public static class PacksField {
```

Comment:
This is the same basic structure that Run History Plus follows. Add a new field, gather data for it, let it get automatically loaded by JSON magic, and patch into the rendering code to display it.